modules/login/php/requestaccount.class.inc

```diff
@@ -135,6 +135,12 @@ class RequestAccount extends \NDB_Form
             $errors['from'] = "Please provide a valid email!";
         } else if (!filter_var($_REQUEST['from'], FILTER_VALIDATE_EMAIL) ) {
             $errors['from'] = 'Please provide a valid email!';
+        } else if (preg_match('[<|>|"|&]', $_REQUEST['from'])) {
+            // Although some of these characters are legal in emails, due to the
+            // current HTML escaping method, it is better to reject email
+            // addresses containing them
+            $errors['from'] =  'Email address can not contain any the following' .
```

Comment:
This is rarely getting triggered because of other validation. `<` is giving the error "You can't use tags in from" from line 165. `"` is giving  the error "You can't use quotes in from" from the same part of the code. `>` is failing with "Please provide a valid email" from line 137.

But there's a bigger problem with quotes in that everything after the `"` gets truncated. I think there's still a reflected XSS attack despite this validation because, while the error is getting triggered, smarty is not escaping it when re-putting the user input into the form with the values that were entered by the user.

Reply:
This does catch "&" though.